### PR TITLE
Expose receipt to calling code (bug 1063821)

### PR DESCRIPTION
**Pull request by kumar303**
@muffinresearch r?

### CODE REVIEW

**Comment by kumar303** [comment]
The downside to the ``!(this instanceof ...)`` in the method is that you have to copy/paste the arguments and someone might forget to update them when the signature changes. I found this ``createReceipt()`` approach instead but it's a little ugly.

**Comment by kumar303** [comment]
note to self: I need to add ``receiptInfo`` to the docs when this is merged.

**Comment by muffinresearch** [comment]
> The downside to the !(this instanceof ...) in the method is that you have to copy/paste the arguments and someone might forget to update them when the signature changes. I found this createReceipt() approach instead but it's a little ugly.

Could you not use `arguments`? Maybe something like this: http://jsbin.com/toguro/2/edit?js,console

File: lib/fxpay/receipts.js [hunk 1]
```diff
@@ -500,4 +500,40 @@
...
+
+  function ReceiptInfo(receipt, verificationResult) {
+    // Information about a receipt.
+
+    // This is the original JWT (JSON Web Token) receipt string.
+    this.receipt = receipt;
+
+    // verificationResult is a JSON server response from
+    // the validator. For more info see:
+    // https://wiki.mozilla.org/Apps
```
Comment (muffinresearch) [inline lib/fxpay/receipts.js]: Probably ignore the 80 chars limit to make this more cut-and-pasteable.

File: lib/fxpay/receipts.js [hunk 2]
```diff
@@ -500,4 +500,40 @@
     return atob(s);
   }
 
+
+  function ReceiptInfo(receipt, verificationResult) {
+    // Information about a receipt.
+
+    // This is the original JWT (JSON Web Token) receipt string.
```
Comment (muffinresearch) [inline lib/fxpay/receipts.js]: Good call on the comments.

File: lib/fxpay/receipts.js [hunk 3]
```diff
@@ -500,4 +500,40 @@
...
+  var createReceiptInfo = (function() {
+
+    function Creator(args) {
+      return ReceiptInfo.apply(this, args);
+    }
+
+    Creator.prototype = ReceiptInfo.prototype;
+
+    return function() {
+      return new Creator(arguments);
```
Comment (muffinresearch) [inline lib/fxpay/receipts.js]: Hah I see this is pretty much what I was talking about earlier.

**Comment by muffinresearch** [comment]
Yeah I think the constructor stuff could maybe be a little bit more self-contained but otherwise lgtm.

r+wc

File: lib/fxpay/receipts.js (hunk 2 above)
Comment (kumar303) [inline lib/fxpay/receipts.js]: Yeah, I also want to make an object like this for ProductInfo which is documented on MDN but not really in the code. I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1143024

**Comment by kumar303** [comment]
Ah, this self-contained constructor approach is way better

**Pull request merged**